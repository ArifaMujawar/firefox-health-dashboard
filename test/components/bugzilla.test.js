import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import BugzillaQueryUrl from '../../src/components/BugzillaUrl';

describe('<BugzillaQueryUrl />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <BugzillaQueryUrl
          bugCount={1}
          url="http://github.com"
          text="test"
          classes={{
            padding: 'X',
            text: 'Y',
            margin: 'Z',
          }}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly without bugs', () => {
    const tree = renderer
      .create(
        <BugzillaQueryUrl
          url="http://github.com"
          text="test"
          classes={{
            padding: 'X',
            text: 'Y',
            margin: 'Z',
          }}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
