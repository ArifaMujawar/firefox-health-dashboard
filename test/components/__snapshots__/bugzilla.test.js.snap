// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BugzillaQueryUrl /> renders correctly 1`] = `
<a
  href="http://github.com"
  rel="noopener noreferrer"
  target="_blank"
>
  <span
    className="MuiBadge-root-3 BugzillaQueryUrl-margin-1 Z"
  >
    <p
      className="MuiTypography-root-10 MuiTypography-body1-19 BugzillaQueryUrl-text-2 Y"
    >
      test
    </p>
    <span
      className="MuiBadge-badge-4 MuiBadge-colorPrimary-5"
    >
      1
    </span>
  </span>
</a>
`;

exports[`<BugzillaQueryUrl /> renders correctly without bugs 1`] = `
<a
  href="http://github.com"
  rel="noopener noreferrer"
  target="_blank"
>
  <p
    className="MuiTypography-root-10 MuiTypography-body1-19 undefined X"
  >
    test
  </p>
</a>
`;
